PhIPSeq_external/Analysis_allergens_and_IEDB/prediction_helpers.py

```diff
@@ -0,0 +1,233 @@
+import os
+import pickle
+import seaborn as sns
+import numpy as np
+import pandas as pd
+import shap
+from matplotlib import pyplot as plt
+from scipy.stats import pearsonr, fisher_exact
+from sklearn.ensemble import GradientBoostingClassifier, GradientBoostingRegressor
+from sklearn.metrics import roc_curve, auc, mean_squared_error, r2_score
+from sklearn.model_selection import KFold, train_test_split
+from statsmodels.stats.multitest import fdrcorrection
+
+from PhIPSeq_external.Analysis_allergens_and_IEDB.config import base_path, out_path, MIN_OLIS
+
+
+def get_oligos():
+    meta = pd.read_csv(os.path.join(base_path, "cohort.csv"), index_col=0, low_memory=False)
+    df = pd.read_csv(os.path.join(base_path, "fold_data.csv"), index_col=[0, 1], low_memory=False).loc[
+        meta.index].unstack()
+    df = df.T.reset_index(level=0, drop=True).T
+    df = df.loc[:, df.min().ne(-1)].copy()
+
+    # Filter to use only IEDB or allergens
+    df_info = pd.read_csv(os.path.join(base_path, "library_contents.csv"), index_col=0, low_memory=False)
+    df_info = df_info[df_info['num_copy'].eq(1)][['is_IEDB', 'is_allergens']]
+    df_info = df_info[df_info.any(axis=1)].index
+    df = df.loc[:, df.columns.isin(df_info)].copy()
+    assert df.min().min() > 1
+    df.fillna(1, inplace=True)
+    df = np.log(df)
+    return df
+
+
+def configure_x_y(x, y, metadata=None, is_classifier=False):
+    y = y.dropna()
+    x = x.loc[x.index.intersection(y.index)]
+    y = y.loc[x.index].copy()
+    if is_classifier:
+        y = y.astype(int)
+
+    # filter oligos that appear in less than 5% of the population
+    x = x.loc[:, x.gt(0).mean().gt(0.05)].copy()
+
+    # Add year of birth and gender
+    if metadata is None:
+        metadata = pd.read_csv(os.path.join(base_path, "cohort.csv"), index_col=0, low_memory=False)[
+            ['yob', 'gender']].dropna()
```

Comment:
1. why not call read_metadata?
2. don;t need to dropna() as it is done in next line anyhow.